packages/toolpad-components/src/Text.tsx

```diff
@@ -23,9 +22,22 @@ interface TextProps extends Omit<BaseProps, 'children'> {
 
 const MarkdownContainer = styled('div')({
   display: 'block',
-  [`&:empty::before`]: { content: '""', display: 'inline-block' },
+  maxWidth: 'fill-available',
+  '& span:empty::before': {
+    content: '""',
+    display: 'inline-block',
+  },
 });
```

Comment:
I think that it would be good to remove the margin top and bottom. It looks like Retool had the same idea:

<img width="175" alt="Screenshot 2023-01-09 at 00 41 50" src="https://user-images.githubusercontent.com/3165635/211224832-7a1a7eda-d511-4e8a-a3bd-a7f8f404f932.png">
<img width="176" alt="Screenshot 2023-01-09 at 00 41 58" src="https://user-images.githubusercontent.com/3165635/211224829-3a6c2746-a990-4eb7-bab9-659e9db39289.png">

The problem right now is that

```
foo

bar
```

renders like this:

<img width="377" alt="Screenshot 2023-01-09 at 00 43 18" src="https://user-images.githubusercontent.com/3165635/211224874-e06b8c1a-747f-439d-9242-75d1cfb93026.png">

Reply:
I don't see any margin on the markdown element: 
<img width="1308" alt="Screenshot 2023-01-09 at 6 25 19 PM" src="https://user-images.githubusercontent.com/19550456/211312927-4dc1c3db-b005-497b-bbef-e1a80a400086.png">

Reply:
I think it's [collapsed](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Box_Model/Mastering_margin_collapsing) from the one in the paragraphs